evm_loader/program/src/evm/database.rs

```diff
@@ -2,6 +2,7 @@ use super::{Buffer, Context};
 use crate::{error::Result, types::Address};
 use ethnum::U256;
 use solana_program::{account_info::AccountInfo, pubkey::Pubkey};
+use std::sync::Arc;
```

Comment:
Don't use `Arc` in evm code. This cause a lot of unnecessary side effects. And atomics are not supported by BPF.

Reply:
Rolled back.